primestg/report/base.py

```diff
@@ -59,12 +58,11 @@ def _get_timestamp(self, name, element=None):
         if date_value.upper() in BAD_TIMESTAMP or not date_value:
             date_value = '19000101000000W'
 
-        if date_value.startswith('ffff'):
-            date_value = date_value.replace(date_value[:4], '9999')
+        if date_value.upper().startswith('FFFF') or e.get('DTCard') == 'Y':
```

Comment:
* For **S23** `SpecialDays`, better do not use` _get_timestamp`
* Add some tests